### PR TITLE
feat(parser): Make `LIKE/ILIKE` as a kind of Expr in ast, and parse ESCAPE

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Prerequisite for #15701 

Most of the codes are ported from the upstream.

Port https://github.com/sqlparser-rs/sqlparser-rs/pull/569
Port https://github.com/sqlparser-rs/sqlparser-rs/pull/1096

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/tests/regress/data/sql/strings.sql [hunk 1]
```diff
@@ -182,8 +182,8 @@ SELECT 'abcdefg' SIMILAR TO '_bcd\%' AS false;
 -- and an empty string to mean "no escape", which is also not per spec
 SELECT 'abcd\efg' SIMILAR TO '_bcd\%' ESCAPE '' AS true;
 -- these behaviors are per spec, though:
-SELECT 'abcdefg' SIMILAR TO '_bcd%' ESCAPE NULL AS null;
-SELECT 'abcdefg' SIMILAR TO '_bcd#%' ESCAPE '##' AS error;
+--@ SELECT 'abcdefg' SIMILAR TO '_bcd%' ESCAPE NULL AS null;
+--@ SELECT 'abcdefg' SIMILAR TO '_bcd#%' ESCAPE '##' AS error;
```
Comment (wangrunji0408) [inline src/tests/regress/data/sql/strings.sql:186]: Why are these tests disabled?

**Review by wangrunji0408 — APPROVED** [review]
Generally LGTM!

File: src/tests/regress/data/sql/strings.sql (hunk 1 above)
Comment (stdrc) [reply src/tests/regress/data/sql/strings.sql:186]: So why?

Comment (TennyZhuang) [reply src/tests/regress/data/sql/strings.sql:186]: Forget to support NULL as escape char, the output is always NULL. I guess it's not important, I'll fix it later.

Comment (TennyZhuang) [reply src/tests/regress/data/sql/strings.sql:186]: The behavior is confusing. Escape ‘’ means no escape, and escape NULL results NULL output.

**Pull request merged**